arch/arm/core/cortex_m/irq_manage.c

```diff
@@ -132,33 +132,17 @@ void z_irq_spurious(const void *unused)
 #ifdef CONFIG_PM
 void _arch_isr_direct_pm(void)
 {
-#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
 	unsigned int key;
 
-	/* irq_lock() does what we want for this CPU */
+	/* Disable all interrupts except ZLIs. */
 	key = irq_lock();
```

Comment:
It doesn't really matter in context, but while you're here it's probably best to swap this for `arch_irq_lock()`, which is the same thing on any uniprocessor architecture.  But on SMP, "irq_lock()" is actually a complicated global spinlock and "arch_irq_lock()" is the call that locally masks interrupts.